### PR TITLE
Upgraded to go 1.17

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Upgraded to go 1.17 as 1.14 is not supported anymore.
- Pinned the docker file to a hash https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies

### CODE REVIEW

**Comment by qustavo** [comment]
Thanks, could you update the release too: https://github.com/mempool/mempool-cli/blob/master/.github/workflows/release.yml#L22 ?

**Comment by naveensrinivasan** [comment]
> Thanks, could you update the release too: https://github.com/mempool/mempool-cli/blob/master/.github/workflows/release.yml#L22 ?

I have updated that.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.14-alpine
+FROM golang:1.17-alpine@sha256:b35984144ec2c2dfd6200e112a9b8ecec4a8fd9eff0babaff330f1f82f14cb2a
```
Comment (qustavo) [inline Dockerfile:1]: why the hash?

Comment (naveensrinivasan) [reply Dockerfile:1]: Document on why to pin dependency https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies

**Pull request merged**